lib/graphql-hive/sampler.rb

```diff
@@ -0,0 +1,63 @@
+module GraphQL
+  class Hive
+    # Dynamic sampler for operations reporting
+    class Sampler
+      def initialize(client_sampler, sample_key_generator = nil)
+        if (client_sampler.respond_to?(:call))
+          @sampler = client_sampler
+          @tracked_operations = Hash.new
+          @sample_key_generator = sample_key_generator
+        elsif (client_sampler.is_a?(Numeric))
+          @sample_rate = client_sampler
+        else
+          @sample_rate = 1
+        end
+      end
+
+      def should_include(operation)
```

Comment:
```suggestion
      def sample?(operation)
```

Suggestion to be more Ruby-like. `?` methods indicate they return a boolean.